### PR TITLE
Remove kubelet version test

### PR DESCRIPTION
**Pull request by jrvaldes**
This change removes the e2e test that checks the kubelet version, so that
WMCO e2e test suite does not fail on k8s version bumps. This test was being
used as a signal and we no longer need it as we are proactive about that.

### CODE REVIEW

File: test/e2e/validation_test.go
Comment (sebsoto) [inline test/e2e/validation_test.go:63]: I'd think this should just be removed and not skipped. There are no plans to re-enable this.

**Comment by saifshaikh48** [comment]
/lgtm

I don't mind either way, skipping or removing the test altogether.

File: test/e2e/validation_test.go
Comment (jrvaldes) [reply test/e2e/validation_test.go]: Removed the test and the associated helper functions.

**Review by aravindhp — APPROVED** [review]
Thanks for doing this, @jrvaldes

**Comment by aravindhp** [comment]
/hold

Blocked by SDN-3522

**Comment by dcbw** [comment]
For the Azure failure, it seems that one of the Windows machines couldn't be created:

```
namespaces/openshift-machine-api/pods/machine-api-controllers-7c4b678fd4-kcb5s/machine-controller/machine-controller/logs/current.log:2022-10-22T20:39:22.009621092Z I1022 20:39:22.009598       1 recorder.go:103] events "msg"="CreateError: failed to reconcile machine \"e2e-wm-kwt2h\"s: failed to create vm e2e-wm-kwt2h: failed to create or get machine: compute.VirtualMachinesCreateOrUpdateFuture: asynchronous operation has not completed" "object"={"kind":"Machine","namespace":"openshift-machine-api","name":"e2e-wm-kwt2h","uid":"4a1b1331-fb52-4872-b4d7-12ac3202ff15","apiVersion":"machine.openshift.io/v1beta1","resourceVersion":"37741"} "reason"="FailedCreate" "type"="Warning"
```

**Comment by dcbw** [comment]
For GCP, e2e test succeeded, but deprovision failed with:

```
level=warning msg=failed to delete network ci-op-g4vc6id6-8411c-s248z-network with error: RESOURCE_IN_USE_BY_ANOTHER_RESOURCE: The network resource 'projects/XXXXXXXXXXXXXXXXXXXXXX/global/networks/ci-op-g4vc6id6-8411c-s248z-network' is already being used by 'projects/XXXXXXXXXXXXXXXXXXXXXX/global/firewalls/k8s-fw-a35ad1270d3ed47899bb1f5055b8a947'
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:164","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 1h0m0s timeout","severity":"error","time":"2022-10-22T22:15:18Z"}
```

**Comment by mansikulkarni96** [comment]
It seems like we are hitting https://issues.redhat.com/browse/OCPBUGS-1944 again on GCP.

**Comment by aravindhp** [comment]
/override ci/prow/gcp-e2e-operator

I have reopened OCPBUGS-1944

**Comment by aravindhp** [comment]
/hold

until #1286 and #1287 merges.

**Comment by jrvaldes** [comment]
/test aws-e2e-operator

Looks like the [issue](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1272/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1586047372692033536) with 4.13 release stream is [fixed](https://coreos.slack.com/archives/CFUGK0K9R/p1667211487377619). Re-testing.

**Comment by saifshaikh48** [comment]
Any backporting needed?

**Comment by aravindhp** [comment]
I suggest backporting to at least 4.12 to allow other backports to go through without conflicts

**Pull request merged**
